Finalization of the provisional naming scheme for branches (#37)
opendefensecloud/artifact-conduit: docs/developer-guide/adrs/003-Branch-Naming.md

---
status: in decission
date: 2025-11-14
---

# Name Branches Following a Common Standard

## Context and Problem Statement

In our developing using git as a versioned source control system we're implementing our features, improvements, bugfixes etc. in parallel branches before they'll get merged into the main branch. The names of these branches can lead to a misinerpretation of it's intention. A naming status could help.

## Considered Solutions

### Free Branch Naming

> Branches can be named without any classification. The naming does not need any intention while a helpful name surely is allowed. Examles are

Reviewer: ```suggestion
Branches can be named without any classification. The naming does not need any intention while a helpful name surely is allowed. Examples are
```

- `comment-adding`
- `readme`
- `evaluate-artiface-workflow`

#### Pros

- Quite simple, fast, and free

#### Cons

> - Not allways helpfule (see `readme`)

Reviewer: ```suggestion
- Not always helpful (see `readme`)
```

- Verbs describing the branch on different positions
- No grouping possible

### Detailed Prefixed Branch Names

> Prefixes in full writtend standard terms seperated by a slash follow a well known standard. These prefixis describe, why a branch exists. Behind the slash a small detail is used. Well known prefixes as examples are

Reviewer: ```suggestion
Prefixes in full written standard terms separated by a slash follow a well known standard. These prefixes describe, why a branch exists. Behind the slash a small detail is used. Well known prefixes as examples are
```

- `feature/short-description`
- `bugfix/issue-42`
- `hotfix/memory-overflow`
- `improvement/branch-naming`
- `experiment/iter-usage-in-looping`
- `release/2.1.0`

#### Pros

- Intention of a branch is more clear

> - Grouping makes it easier to recognise parallel work

Reviewer: ```suggestion
- Grouping makes it easier to recognize parallel work
```
